src/data/huskelapp/useOppdaterHuskelapp.tsx

```diff
@@ -0,0 +1,23 @@
+import { useValgtPersonident } from "@/hooks/useValgtBruker";
+import { ISHUSKELAPP_ROOT } from "@/apiConstants";
+import { post } from "@/api/axios";
+import { HuskelappDTO } from "@/data/huskelapp/huskelappTypes";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { huskelappQueryKeys } from "@/data/huskelapp/useGetHuskelappQuery";
+
+export const useOppdaterHuskelapp = () => {
+  const personident = useValgtPersonident();
+  const queryClient = useQueryClient();
+  const path = `${ISHUSKELAPP_ROOT}/huskelapp`;
+  const postHuskelapp = (nyHuskelapp: HuskelappDTO) =>
+    post<HuskelappDTO>(path, nyHuskelapp, personident);
+
+  return useMutation({
+    mutationFn: postHuskelapp,
+    onSuccess: () => {
+      return queryClient.invalidateQueries(
```

Comment:
Når man bruker `return` her så vil mutation være loading til queryen har refetchet og deretter lukkes modalen. Her tenker jeg vi kan droppe return slik at modalen lukkes og så refetches queryen i bakgrunnen. Siden man uansett må åpne modalen igjen for å se oppdatert huskelapp.

Reply:
Ref: https://tkdodo.eu/blog/mastering-mutations-in-react-query#awaited-promises